net/instaweb/rewriter/css_inline_filter.cc

```diff
@@ -138,6 +149,18 @@ void CssInlineFilter::EndElementImpl(HtmlElement* element) {
           "CSS not inlined because media does not match screen", element);
       return;
     }
+
+    // Dont inline if style <link> element is in html body AND
+    // move_css_to_head is not enabled.
+    // This is to maintain w3c validation since style element is
+    // not recommended in html body. Issue fix #1153.
+    if (inlining_not_pedantically_spec_ &&
```

Comment:
I ANDed pedantic filter enable check in StartElementImpl where I was checking whether we are in body. 
Now I am changing the variable to in_body_ and will also move pedantic filter enable check in endElementImpl().